src/character.cpp

```diff
@@ -1258,9 +1258,23 @@ int Character::swim_speed() const
     float hand_bonus_mult = ( usable.test( body_part_hand_l ) ? 0.5f : 0.0f ) +
                             ( usable.test( body_part_hand_r ) ? 0.5f : 0.0f );
 
+    const float bmi = get_bmi();
+    float player_weight_modifier = 1.0f;
+    // TODO: add values for underweight characters
+    if( bmi > character_weight_category::morbidly_obese ) {
```

Comment:
It would be really nice to have this on a curve, as having this as a curve instead of flat increases/cutoffs on certain points would also make it that running right after eating would be less practical than it is.

Reply:
Good point. The only thing we need now is a good formula for calories-to-penalty ratio.